File: src/mod10_design_pattern/lesson8_bridge/java/Account.java
package mod10_design_pattern.lesson8_bridge.java;

//classification:
//1 type: savingAccount, debitAccount
//2 currency: galleon, sickles, knuts

//classes without Bridge: SavingAccountGalleon,SavingAccountSickles,SavingAccountKnuts,
//                DebitAccountGalleon, DebitAccountSickles , DebitAccountKnuts

public abstract class Account {
    protected Currency currency;

    public Account(Currency currency) {
        this.currency = currency;
    }

    public abstract String getAccountType();

    public double getBalanceConvertedToGalleons(double amount) {
        return amount * currency.getExchangeRate();
    }
}

File: src/mod10_design_pattern/lesson8_bridge/kotlin/Account.kt
package mod10_design_pattern.lesson8_bridge.kotlin

//classification:
//1 type: savingAccount, debitAccount
//2 currency: galleon, sickles, knuts

//classes without Bridge: SavingAccountGalleon,SavingAccountSickles,SavingAccountKnuts,
//                DebitAccountGalleon, DebitAccountSickles , DebitAccountKnuts

abstract class Account(
    private val currency: Currency
) {
    abstract fun getAccountType():String

    fun getBalanceConvertedToGalleons(amount: Double) = amount * currency.getExchangeRate()


}